storage/gcsemu/memstore.go

```diff
@@ -149,6 +149,10 @@ func (ms *memstore) Copy(srcBucket string, srcFile string, dstBucket string, dst
 
 func (ms *memstore) Delete(bucket string, filename string) error {
 	if filename == "" {
+		if _, ok := ms.buckets[bucket]; !ok {
+			return os.ErrNotExist
+		}
+
```

Comment:
I think this is a data race, you need to lock `m.mu` to read and delete from `ms.buckets`

Reply:
oh nice catch, Ill fix that